baselines/moon/README.md

```diff
@@ -0,0 +1,136 @@
+* A simple-CNN with a projection head for CIFAR-10
+* A ResNet-50 with a projection head for CIFAR-100.
+  
+****Dataset:**** : This directory includes CIFAR-10 and CIFAR-100. They are partitioned in the same way as the paper. The settings are as follow:
+
+| Dataset | partitioning method |
+| :------ | :---: |
+| CIFAR-10  | Dirichlet with beta 0.5 |
+| CIFAR-100 | Dirichlet with beta 0.5 |
+
+
+****Training Hyperparameters:**** :
+
+warning: The following tables show the default hyperparameters.
+
+| Description | Default Value |
+| ----------- | ----- |
+| number of clients | 10 |
+| number of local epochs | 10 |
+| fraction fit | 1.0 |
+| batch size | 64 |
+| learning rate | 0.01 |
+| mu | 1 |
+| temperature | 0.5 |
+| alg | moon |
+| seed | 0 |
+| service_device | cpu |
+| number of rounds | 100 |
+| client resources | {'num_cpus': 2.0, 'num_gpus': 0.0 }|
+
+## Environment Setup
+
+To construct the Python environment follow these steps:
+
+```bash
+# set local python version via pyenv
+pyenv local 3.10.6
+# then fix that for poetry
+poetry env use 3.10.6
+# then install poetry env
+poetry install
+
+# activate the environment
+poetry shell
+
+```
+
+
+## Running the Experiments
+
+First ensure you have activated your Poetry environment (execute `poetry shell` from this directory). To run MOON on CIFAR-10 (Table 1 of the paper), you should run:
+```bash  
+python -m moon.main --config-name cifar10 
+```
+
+To run MOON on CIFAR-100 (Table 1 of the paper), you should run:
+```bash
+python -m moon.main --config-name cifar100
+```
+
+
+You can also run FedProx on CIFAR-10:
+```bash
+python -m moon.main --config-name cifar10_fedprox
+```
+
+To run FedProx on CIFAR-100:
+```bash
+python -m moon.main --config-name cifar100_fedprox
+```
+
+## Expected Results
+
+You can find the output log in `_static` directory. After running the above commands, you can see the accuracy list at the end of the ouput, which is the test accuracy of the global model. For example, in one running, for CIFAR-10 with MOON, the accuracy after running 100 rounds is 0.7071 (see `_static/cifar10_moon.log`). 
+
+For CIFAR-10 with FedProx, the accuracy after running 100 rounds is 0.6852 (see `_static/cifar10_fedprox.log`). For CIFAR100 with MOON, the accuracy after running 100 rounds is 0.6799 (see`_static/cifar100_moon.log`). For CIFAR100 with FedProx, the accuracy after running 100 rounds is 0.6494. The results are summarized below:
+
+
+|  | CIFAR-10 | CIFAR-100 |
+| ----------- | ----- | ----- |
+| MOON | 0.7107 | 0.6799 |
+| FedProx| 0.6852 | 0.6494 |
+
+You can find the curve comparing MOON and FedProx on CIFAR-10 below.
+
+![](_static/cifar10_moon_fedprox.png)
+
+
+You can tune the hyperparameter `mu` for both MOON and FedProx by changing the configuration file in `conf`.
+
+You can also run the experiments in Figure 8 of the paper. To run MOON on CIFAR-100 with 50 clients (Figure 8(a) of the paper):
```

Comment:
Similarly to my comment above, could put this for Figure 8 as a separate sub-section? Then below, again a side-by-side plot following the style of Figure 6.

Reply:
Hi @jafermarq ,

Thank you! I have addressed other comments. For Figure 8, I'm rerunning the experiments and the experiment will take a long time. I will probably need 1-2 weeks to complete it. I'll upload the results once completed. You may go ahead with the current version if it's ok for you.